src/drivers/intel/cavs/interrupt.c

```diff
@@ -44,6 +44,11 @@
 #include <stdlib.h>
 #include <cavs/version.h>
 
+/*
+ * The level2 handler attempts to try and fairly service interrupt sources by
+ * servicing on first come first served basis. If two or more IRQs arrive at the
+ * same time then they are serviced in order of ascending status bit.
```

Comment:
not get what is the intention here, the original code already service all level2 interrupts at the while loop(in ascending status bit sequence), what more fairly are we asking for? random sequence?

Reply:
Old code would not deal with any new IRQs that arrived during processing whereas this does.